src/services/big-query-client.ts

```diff
@@ -0,0 +1,41 @@
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+import { ResponseCache } from './../utils';
+import { GenericClient } from './generic-client';
+
+export interface EmailList {
+  emails: String[];
+}
+
+export interface BigQueryClient {
+  getEmailsData(idUser: String): Promise<ResultWithoutExpectedErrors<EmailList>>;
+}
+
+export class HttpBigQueryClient extends GenericClient implements BigQueryClient {
+  private responseCache = new ResponseCache();
+
+  private mapEmails(response: any): EmailList {
+    return {
+      emails: response.emails,
+    };
+  }
+  public async getEmailsData(idUser: String): Promise<ResultWithoutExpectedErrors<EmailList>> {
+    try {
+      const { jwtToken } = super.getApiConnectionData();
+      const response = await this.axios.request({
+        method: 'GET',
+        url: `/big-query/emails`,
+        params: { idUser: idUser },
+        headers: { Authorization: `token ${jwtToken}` },
+      });
+
+      return this.responseCache.getCachedOrMap(this.getEmailsData, response, (r) => {
+        return r.data && r.data.length
+          ? { success: true, value: r.data.map(this.mapEmails) }
+          : { success: true, value: [''] };
```

Comment:
I am not sure if using `responseCache` is useful here.

The idea behind this is to use the same object's reference when the response is the same.

It will work, but I think that it is not necessary here. I have no a strong opinion about it.

Reply:
the cache was removed